crates/trade/src/lib.rs

```diff
@@ -42,44 +41,31 @@ pub struct MatchParams {
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct TradeParams {
-    /// Our identity
-    pub pubkey: PublicKey,
-    /// Identity of the trade's counterparty
+pub struct Trade {
+    /// The identity of the trading party either maker or taker.
+    pub pub_key: PublicKey,
+
+    /// The leverage of the trading party either maker or taker.
+    pub leverage: u8,
+
+    /// The direction of the trading party either long or short.
     ///
-    /// The identity of the trading party that eas matched to our order by the orderbook.
-    pub pubkey_counterparty: PublicKey,
+    /// Note, this needs to be reversed for the counter party.
+    pub direction: Direction,
 
     /// The id of the order
     ///
     /// The order has to be identifiable by the client when returned from the orderbook, so the
     /// client is in charge of creating this ID and passing it to the orderbook.
-    pub order_id: Uuid,
-
-    /// The orderbook id of the counterparty order
-    ///
-    /// The orderbook id of the order that was matched with ours.
-    /// This can be used by the coordinator to make sure the trade is set up correctly.
-    pub order_id_counterparty: String,
-
-    /// The contract symbol for the order
-    pub contract_symbol: ContractSymbol,
-
-    /// Our leverage
-    ///
-    /// This has to correspond to our order's leverage.
-    pub leverage: f64,
-
-    /// The leverage of the order that was matched
-    ///
-    /// This is the leverage of the counterparty.
-    /// This can be used by the coordinator to make sure the trade is set up correctly.
-    pub leverage_counterparty: f64,
+    pub order_id: i32,
+}
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Match {
     /// The quantity to be used
     ///
     /// This quantity may be the complete amount of either order or a fraction.
-    pub quantity: f64,
+    pub quantity: u64,
```

Comment:
This quantity needs to support fractional amounts, see comment here: [`ab06660` (#244)](https://github.com/get10101/10101/pull/244/commits/ab06660ec958ee826ba9e018dcf7163c3ae939bd#r1131880986)